cluster/src/main/java/org/apache/iotdb/cluster/server/member/DataGroupMember.java

```diff
@@ -890,4 +988,57 @@ public boolean isUnchanged() {
   public void setUnchanged(boolean unchanged) {
     this.unchanged = unchanged;
   }
+
+  public void setAndSaveLastAppliedPartitionTableVersion(long version) {
+    lastAppliedPartitionTableVersion.setVersion(version);
+    lastAppliedPartitionTableVersion.save();
+  }
+
+  private class LastAppliedPatitionTableVersion {
+
+    private static final String VERSION_FILE_NAME = "LAST_PARTITION_TABLE_VERSION";
+
+    private long version = -1;
+
+    private String filePath;
+
+    public LastAppliedPatitionTableVersion(String memberDir) {
+      this.filePath = memberDir + File.separator + VERSION_FILE_NAME;
+      load();
+    }
+
+    private void load() {
+      File versionFile = new File(filePath);
+      if (!versionFile.exists()) {
+        return;
+      }
+      try (FileInputStream fileInputStream = new FileInputStream(filePath);
+          DataInputStream dataInputStream = new DataInputStream(fileInputStream)) {
+        version = dataInputStream.readLong();
+      } catch (Exception e) {
+        logger.warn("Cannot deserialize last partition table version from {}", filePath, e);
+      }
+    }
+
+    public synchronized void save() {
```

Comment:
this should be atomic? maybe we need a atomic-rename mechanism

Reply:
Maybe unnecessary, I implemented it in imitation of other places, like `SlotManger`.